src/mlpack/methods/lmnn/lmnn_function_impl.hpp

```diff
@@ -82,10 +82,33 @@ double LMNNFunction<MetricType>::Evaluate(const arma::mat& transformation)
   // Apply metric over dataset.
   transformedDataset = transformation * dataset;
 
-  if (iteration++ % range == 0)
+  double transformationDiff = 0;
+  if (transformationOld.n_elem != 0 && iteration % range == 0)
   {
+    // Calculate norm of change in transformation.
+    transformationDiff = arma::norm(transformation - transformationOld);
+
+    std::vector<size_t> tempPoints;
+
+    for (size_t i = 0; i < dataset.n_cols; i++)
+    {
+      if (transformationDiff * (2 * norm(i) + norm(impostors(k - 1, i)) +
+          norm(impostors(k, i))) > distance(k, i) -  distance(k - 1, i))
+      {
+        tempPoints.push_back(i);
+      }
+    }
+
+    arma::uvec points = arma::conv_to<arma::uvec>::from(tempPoints);
```

Comment:
Another potential opportunity for speedup would be to allocate a "large-enough" array here, which could be done in a couple ways (I am not sure which would be better, feel free to try each!):

1. Allocate `points` to have length equal to `dataset.n_cols`, then simply fill it and track how full it is.  Then have `Impostors()` also take a parameter indicating how many elements of `points` should be used.

2. Allocate `points` as above, but with length `dataset.n_cols * 0.75` (or some tunable parameter, I am not sure which will be best).  If, during the pruning check process, `points` becomes full, then simply shortcut and run `Impostors()` on everything.

3. Take a two-pass approach, where in the first pass you compute whether or not the point is pruned, then allocate the array with the correct size.  In the second pass, fill the elements.  I'm hesitant to say this will be faster here, but it could be.  The `conv_to<>::from()` operation could take a little time.